### PR TITLE
[DVDDemux] Add new audio codec descriptions

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

This change extend the codec descriptions with the most common audio standard format tyoes used most of the time on all video services

Currently video services using common protocols mainly HLS / DASH can provide videos with more audio tracks with same languages but different codec type, this results that on GUI audio track list appears all the same and its not possible distinguish them, with following change we can show the appropriate audio codec description.

This is done by parsing the _full codec string_ that can be read (when provided) from the existing `CDemuxStream::codecName` variable, e.g. "mp4a.40.2"

The full codec string ususally consist in the fourcc+profile/object type as RFC 6381,
some common definitions can be found also on:
https://dashif.org/codecs/audio/
https://developer.mozilla.org/en-US/docs/Web/Media/Formats/codecs_parameter#mpeg-4_audio
other documents say the same

"JOC" its the specific ec3 atmos format, im not aware of a full codec string that expose it, but i know that exists the mimetype (eac3-joc) for example on DASH protocol its set with a separate xml property that we can translate it in a full codec string
despite not exists a full codec string, it would be great if we could still keep it, since we have the possibility to parse it easily

BEFORE:
immagine

AFTER:
immagine

**UPDATE:**
I have add missing ffmpeg profile cases so that also local files can benefit from this change

SIDE NOTE ~TODO~:
For InputStream addons to set audio profiles is required PR #23541

## Motivation and context

Fix issue https://github.com/xbmc/inputstream.adaptive/issues/650

## How has this been tested?

Play this STRM, required inputstream.adaptive:
```
#KODIPROP:inputstream=inputstream.adaptive
#KODIPROP:inputstream.adaptive.manifest_type=hls
https://devstreaming-cdn.apple.com/videos/streaming/examples/adv_dv_atmos/main.m3u8
```
and tested with a local mp4 video with AAC LC

## What is the effect on users?

user can distinguish audio tracks by format type
## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by lrusak** [comment]
I don't have much knowledge here but it seems like there should be a way to get the subtype rather than just parsing a string. Does FFmpeg not provide a way to specify say the "AAC audio object type"?

**Comment by CastagnaIT** [comment]
@lrusak i have add missing ffmpeg profiles so this change now works also for local files
i have updated PR description for future todo

File: xbmc/utils/StringUtils.cpp [hunk 1]
```diff
@@ -1889,6 +1889,20 @@ std::string StringUtils::FormatFileSize(uint64_t bytes)
...
+  if (isCaseInsensitive)
+  {
+    auto itStr = std::search(str.begin(), str.end(), keyword.begin(), keyword.end(),
+                             [](unsigned char ch1, unsigned char ch2) {
+                               return std::toupper(ch1) == std::toupper(ch2);
+                             });
+    return (itStr != str.end());
+  }
+
+  return std::strstr(str.data(), keyword.data()) != nullptr;
```
Comment (neo1973) [inline xbmc/utils/StringUtils.cpp:1903]: `std::strstr` expects the strings to be null terminated which `std::string_view` isn't. You can use `std::string_view::find` instead.

Comment (CastagnaIT) [reply xbmc/utils/StringUtils.cpp]: thanks

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.cpp [hunk 2]
```diff
@@ -45,8 +64,48 @@ std::string CDemuxStreamAudio::GetStreamType()
...
+          break;
+        case FF_PROFILE_AAC_SSR:
+          strInfo = "AAC-SSR ";
+          break;
+        case FF_PROFILE_AAC_LTP:
+          strInfo = "AAC-LTP ";
+          break;
+        default:
+        {
+          // Try check by ISO codec full string
```
Comment (fuzzard) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.cpp:89]: Is it maybe worth adding that DashIF url here as a reference point for the codec strings?

Comment (CastagnaIT) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.cpp]: comment changed however its RFC 6381

**Comment by CastagnaIT** [comment]
someone can approve? so i can go on with inputstream pr's

**Review by phunkyfish — APPROVED** [review]

**Pull request merged**